### DISCUSSION TITLE
Is there a priority system for inputs?

### DISCUSSION THREAD

**Discussion by drvelociraptor** [Pyxus/fray#92] [Q&A]
I have two bind actions and a composite input using those too at the same time. The problem is one binded action key is for attack, so when the two keys are pressed together, the attack and the composite action happens at the same time. I believe one deals with this issue with a priority system? So if the composite is executed, the notmal attack would not happen. Is this possible in Fray?

**Comment by Kuenaimaku** [comment]
I haven't done much digging around with the base addon, but in the project I've been working on, I modified the FrayCompoundState so that get_next_transitions returns an array of transitions sorted by their priority.

```gdscript
func sort_transition_priority(a: _Transition, b: _Transition) -> bool:
	if (a.transition.priority > b.transition.priority):
		return true
	return false


func get_next_transitions(from: StringName) -> Array[_Transition]:
	if _ERR_INVALID_STATE(from):
		return []

	var transitions: Array[_Transition]

	for transition in _transitions:
		if transition.from == from:
			transitions.append(transition)

	transitions.append_array(get_next_global_transitions(from))
	transitions.sort_custom(sort_transition_priority)
	return transitions

```

This then guarantees that the first transition that meets all prereqs and advance conditions is the one with the highest priority.

**Reply by arnaucmd** [reply]
Thanks will try this. honestly though I'm thinking on doing my own system bc it seems this project is kind of abandoned. on my game there will be around twenty enemies at the same time and each enemy has it's own FrayInput and sends virtual input to it, just like the player. The problem is this addon seems unoptimized because game drops to 10fps and debugger shows FrayInput.gd is eating like half the time frame. So for now it's unusable... will post the issue just in case.

**Reply by arnaucmd** [reply]
btw i'm the same guy as th op, I just lost myc account hahh
